src/Shared/SFA.DAS.SharedOuterApi/Interfaces/IApiClient.cs

```diff
@@ -18,6 +19,10 @@ public interface IApiClient<T> : IGetApiClient<T>
         Task Put(IPutApiRequest request);
         Task Put<TData>(IPutApiRequest<TData> request);
         Task<ApiResponse<TResponse>> PostWithResponseCode<TResponse>(IPostApiRequest request, bool includeResponse = true);
+        Task<ApiResponse<TResponse>> PostWithResponseCode<TData, TResponse>(IPostApiRequest<TData> request, bool includeResponse = true)
+        {
```

Comment:
@ChrisJWoodcock is there a reason this method throws not implemented exception and not the other methods in this interface?